### PR TITLE
Feature: OpenCannabis

### PR DESCRIPTION
**Pull request by sgammon**
This changeset adopts the upstream schema refactor introducing the new OpenCannabis spec. Like other client libraries, import/type alias paths need to be changed wholesale to move forward with the built copy of the library.

There are still issues with generated code, in that, we cannot apply constructor compliance for services in Swift without a `required` designator being manually placed on each gRPC service class. Also, we need typealiases for nested messages.

Upstream issues:
- grpc/grpc-swift#103 - Service initializer interface compliance
- grpc/grpc-swift#53 - Nested proto message error

Upstream work:
- bloombox/Schema#38 - `feature/opencannabis`

Worklog:
- [x] Sync schema with new branch `feature/opencannabis`
- [x] Adapt typealiases and library objects
- [x] General library cleanup
- [x] File issues with gRPC for upstream problems

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,8 +1,9 @@
 [submodule "SwiftProtobuf"]
 	path = SwiftProtobuf
 	url = https://github.com/apple/swift-protobuf.git
-	branch = master
+	tag = 1.0.2
 [submodule "SwiftGRPC"]
 	path = SwiftGRPC
 	url = https://github.com/grpc/grpc-swift.git
 	branch = master
+	tag = 0.3.2
```
Comment (sgammon) [inline .gitmodules]: upgrade `SwiftProtobuf` and `SwiftGRPC`

File: Sources/Client/Bindings.swift [hunk 2]
```diff
@@ -15,129 +15,133 @@ import SwiftProtobuf
...
+public typealias VerifyError = Bloombox_Schema_Services_Shop_V1_VerifyError
+public typealias EnrollmentError = Bloombox_Schema_Services_Shop_V1_EnrollmentError
 
 /**
  * Method `ping`.
  */
-public typealias ShopPing = Services_Shop_V1_Ping
-public typealias ShopPingCall = Services_Shop_V1_ShopPingCall
+public typealias ShopPing = Bloombox_Schema_Services_Shop_V1_Ping
+public typealias ShopPingCall = Bloombox_Schema_Services_Shop_V1_ShopPingCall
```
Comment (sgammon) [inline Sources/Client/Bindings.swift]: lots of bindings updates due to path changes

File: Sources/Client/ShopClient.swift [hunk 3]
```diff
@@ -73,8 +68,8 @@ public final class ShopClient: RemoteService {
   /**
    * Shop service.
    */
-  internal func service(_ apiKey: APIKey) -> Shop {
-    let svc = RPCServiceFactory<Shop>.factory(forService: Transport.config.services.shop)
+  internal func service(_ apiKey: APIKey) -> ShopService {
+    let svc = RPCServiceFactory<ShopService>.factory(forService: Transport.config.services.shop)
```
Comment (sgammon) [inline Sources/Client/ShopClient.swift]: some basic consistency changes here - we have `MenuService`, we should also have `ShopService`

File: Sources/Schema/analytics_commerce_ProductAnalytics.pb.swift [hunk 4]
```diff
@@ -94,12 +94,12 @@ public enum Analytics_Product_ProductAction: SwiftProtobuf.Enum {
...
 /// for detail information to an end user.
-public struct Analytics_Product_Impression: SwiftProtobuf.Message {
+public struct Bloombox_Schema_Analytics_Product_Impression: SwiftProtobuf.Message {
   public static let protoMessageName: String = _protobuf_package + ".Impression"
 
   /// Key for the product that was viewed.
-  public var key: Base_ProductKey {
-    get {return _storage._key ?? Base_ProductKey()}
+  public var key: Opencannabis_Base_ProductKey {
+    get {return _storage._key ?? Opencannabis_Base_ProductKey()}
```
Comment (sgammon) [inline Sources/Schema/analytics_commerce_ProductAnalytics.pb.swift]: schema updates here illustrate the compiled code changes

File: Sources/Client/ShopClient.swift (hunk 3 above)
Comment (tyler-porras) [reply Sources/Client/ShopClient.swift]: Perfect is the enemy of done.

**Pull request merged**